### PR TITLE
Fix godoc badge image

### PR DESCRIPTION
**Pull request by abhinav**
pkg.go.dev does not serve a badge like godoc.org. This change switches
the image over to one provided by shields.io.

Note that this uses the style "for-the-badge" from shields.io.

---

Preview

image

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
                             </td>
                             <td>
                                 <a href="//{{ $.Godoc.Host }}/{{ $importPath }}">
-                                    <img src="//{{ $.Godoc.Host }}/{{ $importPath }}?status.svg" alt="GoDoc" />
+                                    <img src="//img.shields.io/badge/godoc-reference-blue?style=for-the-badge" alt="GoDoc" />
```
Comment (r-hang) [inline templates/index.html]: Does going to: https://img.shields.io/badge/godoc-reference-blue%253Fstyle=for-the-badge yield the badge that we want?

Comment (r-hang) [reply templates/index.html]: Yes, it does!

File: go.mod [hunk 2]
```diff
@@ -1,5 +1,7 @@
 module go.uber.org/sally
 
+go 1.14
```
Comment (SyrieBianco) [inline go.mod]: Just double checking if we want to specify the go version here and thereby change the checksum for go-bindata below

Comment (abhinav) [reply go.mod]: Fair point. This probably isn't necessary and probably should be done separately.

Comment (abhinav) [reply go.mod]: I'm seeing a checksum mismatch with 1.12 as well as 1.13.
I'm gonna guess go-bindata made this change.
I'll still omit the 1.14 change from this diff.

**Pull request merged**